src/params.rs

```diff
@@ -214,4 +214,23 @@ pub mod repos {
         Pushed,
         FullName,
     }
+
+    /// A Git reference, either a branch or a tag.
+    pub enum Reference {
+        Branch(String),
+        Tag(String),
+    }
+
+    impl Reference {
+        pub fn ref_url(&self) -> String {
+            match self {
+                Self::Branch(branch) => format!("heads/{}", branch),
+                Self::Tag(tag) => format!("tags/{}", tag),
+            }
+        }
+
+        pub fn full_ref_url(&self) -> String {
```

Comment:
Instead of or in addition to this method, I would add a `Display` impl that's equivalent.